src/Visualization/Python/PlotPowerMonitors.py

```diff
@@ -0,0 +1,312 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import fnmatch
+import logging
+import os
+from itertools import cycle
+from typing import Iterable, Optional, Sequence, Union
+
+import click
+import matplotlib.pyplot as plt
+import numpy as np
+import spectre.IO.H5 as spectre_h5
+from spectre.DataStructures import DataVector
+from spectre.Domain import Domain, deserialize_domain
+from spectre.IO.H5.IterElements import iter_elements
+from spectre.NumericalAlgorithms.LinearOperators import power_monitors
+
+logger = logging.getLogger(__name__)
+
+
+def find_block_or_group(
+        block_id: int, block_or_group_names: Sequence[str],
+        domain: Union[Domain[1], Domain[2], Domain[3]]) -> Optional[int]:
+    """Find entry in 'block_or_group_names' that corresponds to the 'block_id'
+    """
+    block_name = domain.blocks[block_id].name
+    for i, name in enumerate(block_or_group_names):
+        if name == block_name:
+            return i
+        if (name in domain.block_groups
+                and block_name in domain.block_groups[name]):
+            return i
+    return None
+
+
+def plot_power_monitors(volfiles: Union[spectre_h5.H5Vol,
+                                        Iterable[spectre_h5.H5Vol]],
+                        obs_id: int, tensor_components: Sequence[str],
+                        block_or_group_names: Sequence[str],
+                        domain: Union[Domain[1], Domain[2], Domain[3]]):
+    # One subplot per block or group
+    num_plots = len(block_or_group_names)
+    fig, axes = plt.subplots(nrows=1,
+                             ncols=num_plots,
+                             figsize=(num_plots * 4, 4),
```

Comment:
If you have more than two plots, this seems like it'll just create a super wide image/pdf. Is that intended? What about rows of 2 plots?

Reply:
You choose the block groups on the command line, so you probably won't make plots with more than 3 or 4 block groups. Also, the single row helps comparing the truncation errors side-by-side. So I'd keep the single row for now. If this becomes an issue we can think of ways to organize the plot layout better.